src/tools/create-http-patch.js

```diff
@@ -0,0 +1,69 @@
+const _ = require('lodash');
+const constants = require('../constants');
+
+const createHttpPatch = (event) => {
+  const createLogger = require('./create-logger');
+  const logBuffer = [];
+  const logger = createLogger(event, {logBuffer});
+
+  const httpPatch = (object) => {
+    const originalRequest = object.request;
+
+    // Proxy the request method
+    object.request = (options, callback) => {
+      const requestUrl = options.url || options.href || (`${options.protocol || 'https://'}${options.host}${options.path}`);
+      const logger_url = process.env.LOGGING_ENDPOINT || constants.DEFAULT_LOGGING_HTTP_ENDPOINT;
+
+      // Ignore logger requests
+      if (requestUrl.indexOf(logger_url) !== -1) {
+        return originalRequest(options, callback);
+      }
+
+      // Ignore requests made via the request client
+      if (_.get(options.headers, 'user-agent', []).indexOf('Zapier') !== -1) {
+        return originalRequest(options, callback);
+      }
+
+      // Proxy the callback to get the response
+      const newCallback = function(response) {
+        const requestHeaders = _.map(Object.keys(options.headers || {}), (header) => `${header}: ${options.headers[header]}`);
+        const responseHeaders = _.map(Object.keys(response.headers || {}), (header) => `${header}: ${response.headers[header]}`);
+        const responseBody = [];
+
+        const logResponse = () => {
+          // Prepare data for GL
+          const logData = {
+            log_type: 'http',
+            request_type: 'devplatform-outbound',
+            request_url: requestUrl,
+            request_method: options.method || 'GET',
+            request_headers: requestHeaders.join('\n'),
+            request_data: options.body || '',
```

Comment:
Might make sense to check if this was a file and do something like <probably binary data> instead of the full body.

Reply:
I didn't want to do it here because the `createLogger` takes care of truncating and similar "cleaning" (like obfuscating sensitive data), so it'd effectively be almost duplicating work in a very quick sequential way.